adapter/internal/adapter/adapter.go

```diff
@@ -188,7 +183,13 @@ func Run(conf *config.Config) {
 	// Set enforcer startup configs
 	xds.UpdateEnforcerConfig(conf)
 
-	go operator.InitOperator()
+	go operator.InitOperator(conf.Adapter.Metrics.Port)
+
+	// Start the metrics server
+	if conf.Adapter.Metrics.Enabled && strings.EqualFold(conf.Adapter.Metrics.Type, metrics.PrometheusMetricType) {
+		loggers.LoggerAPKOperator.Info("Starting Prometheus Metrics Server ....")
+		go metrics.StartPrometheusMetricsServer()
```

Comment:
Shall we rename this method StartPrometheusMetricsServer? This only registers the collector right?

Reply:
Fixed in [18fc290](https://github.com/wso2/apk/pull/2135/commits/18fc290c1e2e64c42ef33ff55697a3d9139bb8bd)